### PR TITLE
Add description of local kw in global scope

### LINKED ISSUE #42664: Make documentation clear about handling of `local` in global scope
Following https://stackoverflow.com/questions/69587809/how-to-think-of-begin-block-in-julia and discussion on Slack it would be ideal to make a clear explanation in the Julia manual how the following code blocks:
```
x = 1
local x = 0
@show x
```
and
```
x = 1
begin
    local x = 0
    @show x
    x += 1
end
@show x
```
and
```
x = 1
if true
    local x = 0
    @show x
    x += 1
end
@show x
```
are handled if they are in global scope in two modes:
1. run in REPL
2. run from script

CC @JeffBezanson

### PR DESCRIPTION
**Pull request by tomtuamnuq**
Closes #42664 
Adds an explanation of `local` keyword usage in a top-level expression to the Julia Manual. 
It contains the description @JeffBezanson gave in #10472 and an example as @bkamins gave in #42664.

### CODE REVIEW

**Comment by tomtuamnuq** [comment]
Think it is ready to merge.  It just adds an example with explanation regarding the scope to the docs.

@bkamins

**Comment by bkamins** [comment]
@tomtuamnuq - I am not a JuliaLang member. I do not have rights to approve nor merge PRs here. We need to wait for some of the core developers to review this PR.

**Comment by StefanKarpinski** [comment]
I think that @JeffBezanson needs to review this but it looks ok to me.

File: doc/src/manual/variables-and-scoping.md
Comment (simeonschaub) [inline doc/src/manual/variables-and-scoping.md:102]: > Each top-level expression has an invisible scope block around it.

I think that statement is a bit confusing, since toplevel expressions don't introduce an actual scope block like `let` or `while` where new assignments are always scoped by default. We should make clear that this only affects variables with an explicit `local` declaration.

Comment (StefanKarpinski) [reply doc/src/manual/variables-and-scoping.md:102]: This is why I don't feel like I can clearly explain what's happening here. Is there a scope or not? If there isn't a scope, then what is the variable local to? Maybe there's a way to explain it, but if so, I don't know how to.

Comment (simeonschaub) [reply doc/src/manual/variables-and-scoping.md:102]: I think I would mostly leave out the word scope here and just say that `local` inside top-level expressions means that the variable won't be accessible outside that expression.

Comment (tomtuamnuq) [reply doc/src/manual/variables-and-scoping.md]: @simeonschaub Since the usage of scope seems to be confusing here I removed it.

File: doc/src/manual/variables-and-scoping.md
Comment (simeonschaub) [inline doc/src/manual/variables-and-scoping.md:104]:
```suggestion
The variable inside the expression does not affect global variables of the same name.
```

**Review by simeonschaub — APPROVED** [review]
Thanks! One small tweak, otherwise this looks good to me.

**Comment by simeonschaub** [comment]
Ah, and it seems like you need to fix the whitespace.

**Comment by simeonschaub** [comment]
Doctest hang is likely unrelated, but let's rerun it just to be safe.

**Comment by tomtuamnuq** [comment]
@simeonschaub Thanks for your guidance. tester_linux64 failed again. However, I have not found something about it in open issues.

**Comment by simeonschaub** [comment]
That one passed before, I believe that hang is a known issue with CI. Definitely nothing that would be caused by this PR.

**Pull request merged**
